### PR TITLE
Add placement group support for nodegroups

### LINKED ISSUE #479: Placement Group support when creating EKS cluster
*Before creating a feature request, please search existing feature requests to see if you find a similar one. If there is a similar feature request please up-vote it and/or add your comments to it instead*
I didn't find it's there, please correct me if you already have it. 

**Why do you want this feature?**
Lots of machine learning workloads requires low latency network support and users like to leverage placement group feature to achieve that. (I think single AZ is supported and we'd like to move further)

**What feature/behavior/change do you want?**
A clear and concise description of the change you would like to see in eksctl.
1. Add placement group option
2. support all placements
3. Add into cloudFormation template
4. do basic validation, for example, cluster placement group has to be in single AZ. has to get at least 2 AZ to support spread. 

```
eksctl create cluster --region=us-west-2 --placement_group=cluster --xxxxx
```

Do not hesitate, when appropriate, to share the exact commands or API you would like, and/or to share a diagram (e.g.: asciiflow.com): "a picture is worth a thousand words".

Let me know this this is reasonable and I can help on the implementation

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description
Closes #479 

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
  - Added to schema
- [x] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/cfn/builder/managed_launch_template.go [hunk 1]
```diff
@@ -120,6 +120,12 @@ func (m *ManagedNodeGroupResourceSet) makeLaunchTemplateData() (*gfnec2.LaunchTe
 		launchTemplateData.BlockDeviceMappings = []gfnec2.LaunchTemplate_BlockDeviceMapping{mapping}
 	}
 
+	if mng.Placement != nil {
```
Comment (martina-if) [inline pkg/cfn/builder/managed_launch_template.go:123]: Can we add a unit test for these?

Comment (michaelbeaumont) [reply pkg/cfn/builder/managed_launch_template.go:123]: done

**Review by martina-if — APPROVED** [review]

**Comment by jarednielsen** [comment]
Does this work for creating a new placement group, or only for existing ones?

**Comment by michaelbeaumont** [comment]
@jarednielsen It's only for referencing existing placement groups at the moment.

**Pull request merged**